ls1intum/Artemis
# [Enhancement] Improve user interface for Manual Assessment of Programming Exercises

### Checklist

### Motivation and Context


We want to improve the UI of the manual assessment of programming exercises and align it more to the other exercises.

### Description

Comment from @krusche in #2223 : 
- [x] Only show 1 result (top right corner) at the same time based on the last automatic one, that automatically adapts to the manual assessment input. This means it should also be clickable and show the current manual and automatic feedback in the popup
- [x] Remove the Grading Instructions below the code editor
- [x]  Move the button Download repos to the upper right corner left to the result, rename it to "Download repo" on this page, because only one single repository will be downloaded
- [x]  Also move the link "Go to repository" to the upper right corner left to the result
- [x] The result should show the current score in percentage (in addition it can show the score in points in case this is helpful, but then both needs to be in sync)
- [ ]  In the exercise dashboard, change the buttons "New Result" and "Update result" to how they are named in other exercise types. This has to be consistent. -> Follow-up PR

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create programming exercise with SCA and without SCA
4. Participate, Assess, Complain and Assess the Complain (everything should work as intended, try out different scenarios)
5. Try also out the Configure Grading see PR test steps of #2223 (steps for programming exercise)

### Test Coverage



### Screenshots



Screenshots are made with a SCA Programming Exercise with 100 maxScore and 10 bonus points, and deduction for code style issues each 1 point.

**NOTE**: Do not wonder why the issues are not listed in the result string before the configure grading test case change. I used submissions, where adding the issue numbers into the result string, was not yet implemented.

New layout of assessment view:
<img width="1413" alt="image" src="https://user-images.githubusercontent.com/41108487/97499619-767e0780-196e-11eb-94d2-d4575df518b1.png">
<img width="1402" alt="image" src="https://user-images.githubusercontent.com/41108487/97499629-7d0c7f00-196e-11eb-8194-7feec5762402.png">

When changing the feedback elements, result string adapts automatically (did not press on save or submit)
<img width="1437" alt="image" src="https://user-images.githubusercontent.com/41108487/97499814-d83e7180-196e-11eb-8e29-2725ea56dda9.png">

Also displayed in the dialog:
<img width="1234" alt="image" src="https://user-images.githubusercontent.com/41108487/97499866-e8eee780-196e-11eb-9e8b-3fc3666ebcf1.png">

After clicking on save the preliminary assessment is visible. (still not visible for students)
<img width="1360" alt="image" src="https://user-images.githubusercontent.com/41108487/97500115-53078c80-196f-11eb-92bb-101b93e3543f.png">

After submitting (tutor view):
<img width="722" alt="image" src="https://user-images.githubusercontent.com/41108487/97500220-81856780-196f-11eb-847a-5face78d3e8b.png">

Student sees the new result string:
<img width="1421" alt="image" src="https://user-images.githubusercontent.com/41108487/97500356-b8f41400-196f-11eb-8efa-c8d9d0759abc.png">

After Changing configure grading (1 extra point for correct bubbleSortTest), the result string adapts accordingly:
NOTE: Now the issues are listed in the result string as well (were not before because of 'old' submission)
![image](https://user-images.githubusercontent.com/41108487/97500842-9adae380-1970-11eb-8001-40faaac3e522.png)
<img width="1427" alt="image" src="https://user-images.githubusercontent.com/41108487/97500871-a7f7d280-1970-11eb-9f99-aa87b7cf8f15.png">

After student complaint and assessing the complaint (adding 10 points in inline feedback and 10 points unreferenced)
![image](https://user-images.githubusercontent.com/41108487/97503021-8e588a00-1974-11eb-9454-b9d8af4ed0b0.png)

![image](https://user-images.githubusercontent.com/41108487/97502972-7a148d00-1974-11eb-8cfd-62737cb6db90.png)

![image](https://user-images.githubusercontent.com/41108487/97503060-9c0e0f80-1974-11eb-955b-e75c0344e79c.png)

Student view after assessing the complaint:
<img width="1437" alt="image" src="https://user-images.githubusercontent.com/41108487/97503168-d37cbc00-1974-11eb-93e4-1b40a4230f19.png">
<img width="1392" alt="image" src="https://user-images.githubusercontent.com/41108487/97503183-dd062400-1974-11eb-9876-fc7d38b1f8d8.png">
